### PR TITLE
feat: [DHIS2-16293][DHIS2-16294] Category combination in event program forms

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
Tech-summary:
- Added a new section in our DataEntryForms for AOC
- Make Categories in ScopeSelector displayOnly when form is open
- Make selected categories default value in the form, but editable

### CODE REVIEW

**Review by simonadomnisoru — COMMENTED** [review]
It looks good overall 👏 . I only added a few small comments

File: src/core_modules/capture-core/components/ScopeSelector/QuickSelector/Program/CategorySelector.component.js [hunk 1]
```diff
@@ -177,9 +181,14 @@ class CategorySelectorPlain extends React.Component<Props, State> {
                 value={selectedCategoryName}
                 noValueMessage={i18n.t(`Choose a ${passOnProps.category.name}`)}
                 open={this.state.open}
-                setOpen={open => this.setState({ open })}
-                onClearSelectionClick={onClearSelectionClick}
+                setOpen={(open) => {
+                    if (displayOnly) return;
```
[the remark this answers is not in the document]
Comment (eirikhaugstulen) [reply src/core_modules/capture-core/components/ScopeSelector/QuickSelector/Program/CategorySelector.component.js:185]: Hm, they did not seem to think it was a bug, or at least something to spend too much time on. We can raise it again if we wan't, but for now this is the quickest solution I think! I can register it as a ticket perhaps.

File: cypress/e2e/NewPage.feature [hunk 2]
```diff
@@ -231,8 +212,39 @@ Feature: User creates a new entries from the registration page
...
   Then the map opens
 
-  Scenario: New focus area without enrollment 
+  Scenario: New focus area without enrollment
   Given you are in the registration page with Ngelehun CHC org unit selected
   And you select Focus area from the available tracked entity types
   And the user clicks the element containing the text: Choose area on map
   Then the map opens
+
+## New event in program with category combination
```
[the remark this answers is not in the document]
Comment (eirikhaugstulen) [reply cypress/e2e/NewPage.feature:221]: Yeah, we should! I'll try to get some up before merging 👍

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.42,2.41.1,2.40.4,2.39.6 versions

File: cypress/e2e/NewPage.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (eirikhaugstulen) [reply cypress/e2e/NewPage.feature:221]: I added tests for viewing and editing AOC on events. Extracted it into a separate file as there was no existing tests for this page. Can you take a look and see if it makes sense? 😄 We can add more tests at a later stage, but did not want to do it in this PR.

**Review by simonadomnisoru — APPROVED** [review]

**Pull request merged**